src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/FileStorageService.cs

```diff
@@ -42,25 +42,58 @@ public async Task CopyReleaseToPublicContainer(PublishReleaseFilesMessage messag
         {
             var privateContainer =
                 await FileStorageUtils.GetCloudBlobContainerAsync(_privateStorageConnectionString,
-                    PrivateContainerName);
+                    PrivateFilesContainerName);
             var publicContainer =
-                await FileStorageUtils.GetCloudBlobContainerAsync(_publicStorageConnectionString, PublicContainerName);
+                await FileStorageUtils.GetCloudBlobContainerAsync(_publicStorageConnectionString,
+                    PublicFilesContainerName);
 
             var sourceDirectoryPath = AdminReleaseDirectoryPath(message.ReleaseId);
             var destinationDirectoryPath = PublicReleaseDirectoryPath(message.PublicationSlug, message.ReleaseSlug);
 
-            await DeleteFilesAsync(publicContainer, destinationDirectoryPath);
+            await DeleteBlobsAsync(publicContainer, destinationDirectoryPath);
 
             await CopyDirectoryAsyncAndZipFiles(sourceDirectoryPath, destinationDirectoryPath, privateContainer,
                 publicContainer, message);
         }
 
+        public async Task DeleteAllContentAsync()
+        {
+            var publicContainer = await FileStorageUtils.GetCloudBlobContainerAsync(_publicStorageConnectionString,
+                    PublicContentContainerName);
+            await DeleteBlobsAsync(publicContainer, "");
+        }
+
         public IEnumerable<FileInfo> ListPublicFiles(string publication, string release)
         {
-            return FileStorageUtils.ListPublicFiles(_publicStorageConnectionString, PublicContainerName, publication,
-                release);
+            return FileStorageUtils.ListPublicFiles(_publicStorageConnectionString, PublicFilesContainerName,
+                publication, release);
         }
 
+        public async Task MoveStagedContentAsync(ReleaseStatus releaseStatus)
+        {
+            var container =
+                await FileStorageUtils.GetCloudBlobContainerAsync(_publicStorageConnectionString, PublicContentContainerName);
+            
+            var sourceDirectoryPath = PublicContentStagingPath();
+            var sourceDirectory = container.GetDirectoryReference(sourceDirectoryPath);
+            var destinationDirectory = container.GetDirectoryReference("");
+
+            var options = new CopyDirectoryOptions
+            {
+                Recursive = true
+            };
+
+            var allFilesTransferred = new List<CloudBlockBlob>();
+
+            var context = new DirectoryTransferContext();
+            context.FileTransferred += (sender, args) => FileTransferredCallback(sender, args, allFilesTransferred);
```

Comment:
Ooh, what's the `+=` syntax used for here?

Reply:
It's the syntax used to subscribe a method to an event. There is more info on it here
[](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-subscribe-to-and-unsubscribe-from-events)

Reply:
Wow OK cool.  Lots of stuff to learn - thanks for that!